build/lib/test/test.js

```diff
@@ -341,6 +342,154 @@ async function killiOSSimulator() {
 	});
 }
 
+class DeviceTestDetails {
+	/**
+	 *
+	 * @param {string} name device/emulator name (as reported via log output. Defaults to empty string)
+	 * @param {'emulator'|'device'|'simulator'} target
+	 * @param {string} snapshotDir
+	 * @param {Promise[]} snapshotPromises
+	 */
+	constructor(name, target, snapshotDir, snapshotPromises) {
+		this.name = name;
+		this.results = [];
+		// FIXME: These things don't feel right to pass in here.
+		this.target = target;
+		this.snapshotDir = snapshotDir;
+		this.snapshotPromises = snapshotPromises;
+		// set to completed as true by default because for multi-device build we'll end up with default entry that just corresponds to
+		// log output from CLI or not from a specific device. If the device specific test suites finish we want to ignore this one
+		// We'll set to false as soon as we see a version or test start
+		this.completed = true;
+		this.resetTestState();
+	}
+
+	appendStderr(line) {
+		this.stderr += line.trim() + '\n';
+	}
+
+	/**
+	 * 	We saw test end before, but failed to parse as JSON because we got partial output, so continue
+	 * trying until it's happy (and resets sawTestEnd to false)
+	 * @param {string} token
+	 */
+	handleTestContinuation(token) {
+		let modifiedToken = token;
+		let hasAnsi = true;
+
+		// strip off leading log level prefix!
+		if (modifiedToken.startsWith('[INFO]')) {
+			modifiedToken = modifiedToken.substring(8);
+			hasAnsi = false;
+			// colored output?
+		} else if (modifiedToken.startsWith('\u001b[32m[INFO] \u001b[39m')) {
+			// technically we get these characters on a colored [INFO] log prefix:
+			// <Buffer 1b 5b 33 32 6d 5b 49 4e 46 4f 5d 20 1b 5b 33 39 6d 3a 20 08 08 20 08 20>
+			//     \u001b [  3  2  m  [  I  N  F  O  ]  <space>\u001b[39m :  <space><backspace><backspace><space><backspace><space>
+			modifiedToken = modifiedToken.substring(24);
+		}
+
+		if (hasAnsi && this.name) {
+			modifiedToken = modifiedToken.substring(this.name.length + 13); // strip ansi color code, '[, device name, '] ', ansi color code
+		} else {
+			modifiedToken = modifiedToken.substring(this.name.length + 3);
+		}
+
+		// FIXME: IF we see another !TEST_START or !TEST_RESULTS_STOP, can we fail this test and move on?
+		return this.tryParsingTestResult(this.partialTestEnd + modifiedToken);
+	}
+
+	/**
+	 * Handle a new line of output for a given device/emulator
+	 * @param {string} token line of output (raw)
+	 * @param {string} stripped output stripped of ansi colors
+	 * @returns {boolean} true if successfully finished the test suite (completed, may have test failures/errors)
+	 */
+	handleLine(token, stripped) {
```

Comment:
<!--
  0 failure: 
  1 warning:  build/lib/test/te...
  
  
  DangerID: danger-id-Danger;
  File: build/lib/test/test.js;
  Line: 408;
-->

- :warning: build/lib/test/test.js line 408 – 'stripped' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)